internal/testutils/basefee/gen_aux_base_fee_test.go

```diff
@@ -0,0 +1,147 @@
+package basefee
+
+import (
+	"context"
+	"fmt"
+	"github.com/taikoxyz/taiko-client/pkg/sender"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/taikoxyz/taiko-client/bindings"
+	"github.com/taikoxyz/taiko-client/internal/testutils"
+	"github.com/taikoxyz/taiko-client/internal/utils"
+)
+
+type BaseFeeSuite struct {
+	testutils.ClientTestSuite
+
+	baseFee *AuxBaseFee
+
+	lastSyncedBlock uint64
+	gasExcess       uint64
+	config          bindings.TaikoL2Config
+}
+
+func (s *BaseFeeSuite) calc1559BaseFee(_l1BlockID, _parentGasUsed uint64) uint64 {
```

Comment:
This function is aim to simulate [_calc1559BaseFee](https://github.com/taikoxyz/taiko-mono/blob/17abc189ca3d2752beb5400c036a650fd5b9c895/packages/protocol/contracts/L2/TaikoL2.sol#L252).